### PR TITLE
Close resources in catalog properly

### PR DESCRIPTION
**Pull request by yittg**
Some resources in catalog are not closed properly, this change normalize the close of catalog.

### CODE REVIEW

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java [hunk 1]
```diff
@@ -76,7 +76,7 @@
  * objects stored in them to assist with namespace-centric catalog exploration.
  * </p>
  */
-public class NessieCatalog extends BaseMetastoreCatalog implements AutoCloseable, SupportsNamespaces, Configurable {
+public class NessieCatalog extends BaseMetastoreCatalog implements SupportsNamespaces, Configurable {
```
Comment (rdblue) [inline nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:79]: @nastra and @rymurr, is this change okay for Nessie?

Comment (yittg) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:79]: Thanks @rdblue for helping ping, i think it's equivalent?

**Review by nastra — APPROVED** [review]
LGTM

File: nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java (hunk 1 above)
Comment (nastra) [reply nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java:79]: yep that seems ok, thanks for the ping

**Review by kbendick — APPROVED** [review]
LGTM. Thanks for adding this @yittg.

**Comment by yittg** [comment]
ping @rdblue.

**Comment by nastra** [comment]
@rdblue shall we merge this?

**Pull request closed without merging**